pkg/schema/type.go

```diff
@@ -145,6 +160,9 @@ func (t *MapItemType) CheckType(node yamlmeta.TypeWithValues) (chk yamlmeta.Type
 		// A Map must've yielded a non-MapItem which is not valid YAML
 		panic(fmt.Sprintf("MapItem type check was called on a non-MapItem: %#v", node))
 	}
+	if _, anyType := t.ValueType.(AnyType); anyType {
+		return
+	}
```

Comment:
Before the `AnyType` came along, we made "null" a special case (i.e. wrote explicit checks for null like the `if` just below here.

One consequence of that design choice is that it forced checking the _Value_ of a MapItem to be split:
- _sometimes_ occur in `MapItem.Check()` and,
-  _other times_ occur in the `{MapType|ArrayType|ScalarType}.CheckType()`

If we implement `@schema/nullable` as a `Type` then all the checks can occur in that second spot.
This leave the `MapItem.Check()` can be a simple structure check like it is in `ArrayItemType.CheckType()`.

So, while this wasn't a design choice made within _this_ story, we're forced to make matters worse... and this starts a downward spiral unless we jump off, right here.

Nothing is sacred and we're _almost always_ better informed today than we were last go around: let's make the `NullType` so that there's no need to do _any_ explicit check here.